resfulservice/src/graphql/resolver/dataset/mutation.js

```diff
@@ -11,10 +11,17 @@ const datasetMutation = {
       req.logger?.error('[createDatasetId]: User unauthorized');
       return errorFormater('Unauthorized', 401);
     }
-    const datasetId = new DatasetId({ user });
     try {
-      await datasetId.save();
-      return datasetId;
+      // TODO: Find a way to reduce latency here
+      const unusedDatasetId = await DatasetId.findOne({ dataset: [] });
+      if (unusedDatasetId?._id) {
+        req.logger.error('[createDatasetId]: Failed to create. User have unused existing dataset Id');
+        const err = { message: `An unused datasetId - ${unusedDatasetId?._id} exist` };
```

Comment:
line 18: User has unused instead of have unused
line 19: exists instead of exist